tests/register/register.go

```diff
@@ -31,6 +39,33 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
-func Register(tType TestType, t types.Test) {
+func register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
 }
+
+// Registers t for every version, inside the same major version, that is equal to or greater than the specified ConfigMinVersion.
+func Register(tType TestType, t types.Test) {
+	// update confgiVersions with new config versions
+	configVersions := [][]semver.Version{
+		{semver.Version{}}, // place holder
+		{types1.MaxVersion},
+		{types20.MaxVersion, types21.MaxVersion, types22.MaxVersion, types23.MaxVersion},
+	}
+
+	version, semverErr := semver.NewVersion(t.ConfigMinVersion)
+
+	register(tType, t)
+	if semverErr == nil && version != nil && t.ConfigMinVersion != "" {
+		initalVersion := *version
+		for *version != configVersions[version.Major][len(configVersions[version.Major])-1] {
+			t := types.DeepCopy(t)
+
+			version.BumpMinor()
+			version = &configVersions[version.Major][version.Minor]
+
+			t.Name += " " + version.String()
+			t.Config = strings.Replace(t.Config, initalVersion.String(), version.String(), -1)
```

Comment:
This attempts to do a replace against the initial version string, shouldn't it be against `$version`?